### ISSUE TITLE
Publish a changelog and tag releases

### ISSUE THREAD

**Issue by danielcompton** [evanw/node-source-map-support#99]
It would be handy if this project published a changelog and tagged releases so I can easily see what has changed and upgrade to new versions.

**Comment by westy92** [comment]
This is long overdue!  I'm looking to upgrade to v0.5.0 and I have no idea what the changes are!  Having to dig through commits in unacceptable.

**Comment by LinusU** [comment]
@westy92 I'd be happy to accept a pull requests that adds a changelog :)

I have a hard time grasping what you mean by "unacceptable" since you haven't paid anyone to use this, it's just an open source project that a bunch of people work on in their spare time...

**Comment by MikeyBurkman** [comment]
@LinusU Backfilling a changelog is a lot of work, and I understand if the team doesn't want to go back and do that, but it would be super beneficial and not terribly difficult to maintain a changelog  going forward. 

Right now it appears I'm running into a memory leak issue (or something) with migrating from v0.4.18 to the latest v0.5.6 code, but without a changelog, I'm forced to try to dig through commits to get any hint of what might be causing the issue. Having a changelog helps me (the user, who's willing to help out OSS projects) help you (the maintainer).

**Comment by LinusU** [comment]
This is not to say that we don't need a changelog, but in your specific case, I would recommend using `git bisect` to track down the regression. It uses a binary search to find the offending commit, so it's super useful for finding out where some problematic behavior was introduced.

Some quick steps:

1. Clone this repository to your computer
2. Run `npm install` and then `npm link` in the cloned folder
3. In your project, run `npm link source-map-support` which will make your project use the cloned version
4. Back in the cloned project, run `git bisect start`
5. Run `git bisect bad` to mark the current version as broken
6. Run `git bisect good v0.4.18` to indicate what version worked
7. Follow the steps, testing by running the code in your project
8. Find the bad commit
9. Open issue here, or even better; send a pull request 🚀 

Good luck!

**Comment by MikeyBurkman** [comment]
Yeah unfortunately the issue only seems to manifest itself when deployed in our test environment, and requires running for 20+ minutes, so honestly just digging through code is probably just as fast. Anyways, that's off topic for this thread. Will open an issue/PR if I trace the error back to here.

**Comment by westy92** [comment]
@LinusU I apologize, "unacceptable" was a poor word to use. I was just surprised a project of this popularity had no changelog. Thank you for your work on this project; it has helped me and my team immensely! :tada:
